src/main/java/net/openhft/chronicle/queue/util/PretouchUtil.java

```diff
@@ -19,15 +19,46 @@
 package net.openhft.chronicle.queue.util;
 
 import net.openhft.chronicle.core.threads.EventHandler;
+import net.openhft.chronicle.core.threads.InvalidEventHandlerException;
+import net.openhft.chronicle.core.util.ObjectUtils;
+import net.openhft.chronicle.queue.impl.single.Pretoucher;
 import net.openhft.chronicle.queue.impl.single.SingleChronicleQueue;
-import net.openhft.chronicle.queue.internal.InternalPretouchHandler;
+import net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder;
 import org.jetbrains.annotations.NotNull;
 
-public final class PretouchUtil {
+public final class PretouchUtil implements PretoucherFactory {
+    public static final PretoucherFactory INSTANCE;
 
-    private PretouchUtil() {}
+    static {
+        PretoucherFactory instance;
+        try {
+            final Class<?> clazz = Class.forName("software.chronicle.enterprise.queue.pretoucher.EnterprisePretouchUtil");
+            instance = (PretoucherFactory) ObjectUtils.newInstance(clazz);
+            assert SingleChronicleQueueBuilder.areEnterpriseFeaturesAvailable();
```

Comment:
Will not happen unless assertions are enabled - does we depend on this check or is it nice to have?

Reply:
just a nice to have - really for our internal benefit